modules/codegen/test/src/smithy4s/codegen/internals/RendererSpec.scala

```diff
@@ -85,6 +85,144 @@ final class RendererSpec extends munit.FunSuite {
       s"""val underlyingSchema: Schema[Map[String, Int]] = map($keySchemaString, $valueSchemaString)"""
     assert(definition.contains(requiredString))
   }
+  test("enum trait hints should be preserved") {
+    val smithy = s"""
+                    |$$version: "2.0"
+                    |namespace smithy4s
+                    |
+                    |@documentation("this is an enum Suit")
+                    |@enum([
+                    |{
+                    |  value: "DIAMOND",
+                    |  name: "DIAMOND",
+                    |  documentation: "this is a DIAMOND"
+                    |},
+                    |{
+                    |  value: "HAERT",
+                    |  name: "HAERT",
+                    |  documentation: "this is a HAERT",
+                    |  deprecated: true
+                    |},
+                    |{
+                    |  value: "HEART",
+                    |  name: "HEART",
+                    |  documentation: "this is a HEART"
+                    |},
+                    |{
+                    |  value: "CLUB",
+                    |  name: "CLUB"
+                    |},
+                    |{
+                    |  value: "SPADE"
+                    |  name: "SPADE"
+                    |}
+                    |])
+                    |string Suit
+                    |
+                    |""".stripMargin
+    val contents = generateScalaCode(smithy).values
+    val definition =
+      contents.find(content =>
+        content.contains("sealed abstract class Suit") && content.contains(
+          "object Suit extends Enumeration[Suit]"
+        )
+      ) match {
+        case None =>
+          fail(
+            "enum must be rendered as a sealed abstract class and its companion object"
+          )
+        case Some(code) => code
+      }
+    val classDoc = """/** this is an enum Suit
+                     |  * @param DIAMOND
```

Comment:
Random thought: would it be better to generate `@param <variant>` even when there's no doc for variant?